### ISSUE TITLE
UnicodeDecodeError when subprocess outputs non-UTF8

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#3222]
https://crashes.qutebrowser.org/view/3ff95e08, similar to #2302:

```
15:03:44 DEBUG    procs      commands:spawn:1182 Executing youtube-dl with args ['-o', '-', 'https://www.youtube.com/watch?v=KnQaS-uLfWY', '|', 'tee', '~/videos/test.mp4', '|', 'mpv', '-'], userscript=False
[...]
15:03:57 DEBUG    procs      guiprocess:on_finished:98 Process finished with code 1, status 0.
15:03:57 DEBUG    message    message:_log_stack:46 Stack for error message:
  File "/nix/store/rwjvq1rn61l4ba29h38jjb7hakmpd99i-qutebrowser-1.0.2/bin/..qutebrowser-wrapped-wrapped", line 12, in <module>
    sys.exit(main())
  File "/nix/store/rwjvq1rn61l4ba29h38jjb7hakmpd99i-qutebrowser-1.0.2/lib/python3.6/site-packages/qutebrowser/qutebrowser.py", line 186, in main
    return app.run(args)
  File "/nix/store/rwjvq1rn61l4ba29h38jjb7hakmpd99i-qutebrowser-1.0.2/lib/python3.6/site-packages/qutebrowser/app.py", line 138, in run
    ret = qt_mainloop()
  File "/nix/store/rwjvq1rn61l4ba29h38jjb7hakmpd99i-qutebrowser-1.0.2/lib/python3.6/site-packages/qutebrowser/app.py", line 148, in qt_mainloop
    return qApp.exec_()
  File "/nix/store/rwjvq1rn61l4ba29h38jjb7hakmpd99i-qutebrowser-1.0.2/lib/python3.6/site-packages/qutebrowser/misc/guiprocess.py", line 110, in on_finished
    self._what.capitalize(), code))
  File "/nix/store/rwjvq1rn61l4ba29h38jjb7hakmpd99i-qutebrowser-1.0.2/lib/python3.6/site-packages/qutebrowser/utils/message.py", line 58, in error
    stack = traceback.format_stack()
15:03:57 ERROR    message    message:error:63 Command exited with status 1.
[...]
15:03:57 ERROR    misc       crashsignal:exception_hook:216 Uncaught exception
Traceback (most recent call last):
  File "/nix/store/rwjvq1rn61l4ba29h38jjb7hakmpd99i-qutebrowser-1.0.2/lib/python3.6/site-packages/qutebrowser/misc/guiprocess.py", line 113, in on_finished
    stdout = bytes(self._proc.readAllStandardOutput()).decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xd2 in position 27: invalid continuation byte
```

**Comment by gebulmer** [comment]
My first thought with this is that each case of `.decode(foo)` we should use a `try`,  and figure out what the best course of action after is. The browser certainly shouldn't be allowed to just crash!

For the case above, in spawn, I'm assuming non-utf-8 would be valid for passing to the next process?

**Comment by The-Compiler** [comment]
Fixed in #3388.
